server/src/test/java/org/elasticsearch/common/bytes/CompositeBytesReferenceTests.java

```diff
@@ -160,7 +160,25 @@ public void testGetIntLE() {
         assertThat(comp.getIntLE(2), equalTo(0x02010012));
         assertThat(comp.getIntLE(3), equalTo(0x03020100));
         assertThat(comp.getIntLE(4), equalTo(0x04030201));
-        Exception e = expectThrows(ArrayIndexOutOfBoundsException.class, () -> comp.getIntLE(5));
-        assertThat(e.getMessage(), equalTo("Index 4 out of bounds for length 4"));
+        // The jvm can optimize throwing ArrayIndexOutOfBoundsException by reusing the same exception,
+        // but these reused exceptions have no message or stack trace. This sometimes happens when running this test case.
+        // We can assert the exception message if -XX:-OmitStackTraceInFastThrow is set in gradle test task.
+        expectThrows(ArrayIndexOutOfBoundsException.class, () -> comp.getIntLE(5));
```

Comment:
The additional the `testGetDoubleLE()` test sometimes causes the jvm to reuse the same AIOOB exception. These reused exceptions have no message and no stacktrace.

This reproduces when running:

```
./gradlew ':server:test' --tests "org.elasticsearch.common.bytes.CompositeBytesReferenceTests" -Dtests.iters=8
``` 

And also failed in [PR CI](https://gradle-enterprise.elastic.co/s/3ff5rb3m37rdg/tests/:server:test/org.elasticsearch.common.bytes.CompositeBytesReferenceTests/testGetIntLE?top-execution=1).

Running with `OmitStackTraceInFastThrow ` disabled (is enabled by default) stops the exception reuse and test case doesn't fail without this modification:

```
./gradlew ':server:test' --tests "org.elasticsearch.common.bytes.CompositeBytesReferenceTests" -Dtests.iters=8 -Dtests.jvm.argline="-XX:-OmitStackTraceInFastThrow"
```

We either need to ensure `-XX:-OmitStackTraceInFastThrow` is set when running gradle test task or adjust the test, which I have done now. I don't think asserting the exception message is that important?

Reply:
I don't think it is, nah. For what it's worth, painless does set this because it really does want to assert messages. And we set it in production because it can make debugging some issues *impossible*.